### DISCUSSION TITLE
Adding new sensor classes - speed, mass, direction, distance and level

### DISCUSSION THREAD

**Discussion by joakimeriksson** [home-assistant/architecture#605] [Entity Models] [answered]
I did a PR with a set of new sensor classes but realised that there is a need for architectural discussion first. https://github.com/home-assistant/core/pull/54605.

In the PR I added:
     
     DEVICE_CLASS_DIRECTION,
     DEVICE_CLASS_DISTANCE,
     DEVICE_CLASS_FREQUENCY,
     DEVICE_CLASS_LEVEL,
     DEVICE_CLASS_MASS,
     DEVICE_CLASS_PARTICULATE_MATTER_2_5,
     DEVICE_CLASS_PARTICULATE_MATTER_10,
     DEVICE_CLASS_SOUND_PRESSURE,
     DEVICE_CLASS_SPEED

But now PM1, 2.5 and 10 are moving in via another PR. MASS in a previous PR that was very close to being merged (from what I understood). What I would like to do with Home assistant - that is reason for this - is to add complete air quality measuring (maybe we should add RADON also - for Airthings sensors) and add a weather station with wind speed and wind direction. And after having a feature request vote out I got some other suggestions so I decided to make a PR (but failed to do this discussion before).

I am happy to cut-down the PR a bit if some of the sensor classes feel uninteresting or unlikely to be supported by any device - or adapt it in other ways that would be required.

BTW: is there any efforts on refactoring the sensor class model? It feels like there is a lot of repetition of sensor-class lists and constants spread over many files in all the PR:s?

**Comment by frenck** [comment, marked as answer]
We should split this into different topics, which we can discuss individually. Additionally, mass, frequency, mass, and particulate matter already have topics, thus should not be part of this proposal.

**Reply by joakimeriksson** [reply]
Ok - should I just edit this "topic" or create new ones?

**Reply by joakimeriksson** [reply]
Cut out two into this discussion #607

**Reply by andriej** [reply]
Frequency already in #600
